plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/TableStatisticsWriter.java

```diff
@@ -188,6 +284,17 @@ private void copyRetainedStatistics(
         }
     }
 
+    static boolean maySnapshotHaveData(Table table, long snapshotId, FileIO fileIo)
```

Comment:
`snapshotHasData` ?

Reply:
no; the method can return `false` even if snapshot has no data (eg there is an empty data file)

in `snapshotHasData` method, I would have to open such data file and check if empty.
in  `maySnapshotHaveData` i can skip this